### PR TITLE
[ACR] Improve acr login in non-standard consoles and WSL

### PR DESCRIPTION
**Pull request by djyou**
Fixes https://github.com/Azure/azure-cli/issues/5961
Fixes https://github.com/Azure/azure-cli/issues/5729
Fixes https://github.com/Azure/azure-cli/issues/5663

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6136
This is an **experimental preview** for @microsoft users.

**Comment by derekbekoe** [comment]
@sajayantony to review also?

**Comment by sajayantony** [comment]
Does this cover the case for the warning for password and std-in?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -191,46 +179,23 @@ def acr_login(cmd, registry_name, resource_group_name=None, username=None, passw
...
-                           "--password", password,
-                           login_server])
-                p.wait()
+            p = Popen([docker_command, "login",
+                       "--username", username,
+                       "--password", password,
+                       login_server])
+            p.wait()
+        elif b'--password-stdin' in stderr:
+            pass
```
Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py]: The warning on stdin and password is suppressed here. @sajayantony

**Comment by djyou** [comment]
Can we merge this? @derekbekoe @sajayantony

**Pull request merged**